api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +183,219 @@ message JwtHeader {
+//
+// For example:
+//
+// .. code-block:: yaml
+//
+//   providers:
+//      provider1:
+//        issuer: issuer1
+//        audiences:
+//        - audience1
+//        - audience2
+//        remote_jwks:
+//          http_uri:
+//            uri: https://example.com/.well-known/jwks.json
+//            cluster: example_jwks_cluster
+//      provider2:
+//        issuer: issuer2
+//        local_jwks:
+//          inline_string: jwks_string
+//
+//   rules:
+//      # Not jwt verification is required for /health path
+//      - match:
+//          prefix: "/health"
+//
+//      # Jwt verification for provider1 is required for path prefixed with "prefix"
+//      - match:
+//          prefix: "/prefix"
+//        requires:
+//          provider_name: "provider1"
+//
+//      # Jwt verification for either provider1 or provider2 is required for all other requests.
+//      - match:
+//          prefix: "/"
+//        requires:
+//          requires_any:
+//            requirements:
+//              - provider_name: "provider1"
+//              - provider_name: "provider2"
+//
+//// [#not-implemented-hide:]
+message JwtAuthentication {
+  // Map of provider names to JwtProviders.
   //
   // .. code-block:: yaml
   //
-  //   bypass:
-  //   - headers:
-  //     - name: :method
-  //       value: GET
-  //   - headers:
-  //     - name: :path
-  //       exact_match: /healthz
+  //   providers:
+  //     provider1:
+  //        issuer: issuer1
+  //        audiences:
+  //        - audience1
+  //        - audience2
+  //        remote_jwks:
+  //          http_uri:
+  //            uri: https://example.com/.well-known/jwks.json
+  //            cluster: example_jwks_cluster
+  //      provider2:
+  //        issuer: provider2
+  //        local_jwks:
+  //          inline_string: jwks_string
+  //
+  map<string, JwtProvider> providers = 1;
+
+  // Specifies requirements based on the route matches. The first matched requirement will be
+  // applied. If there are overlapped match conditions, please put the most specific match first.
+  //
+  // Examples
+  //
+  // .. code-block:: yaml
   //
-  repeated envoy.api.v2.route.RouteMatch bypass = 3;
+  // rules:
+  //   - match: { prefix: "/healthz" }
+  //   - match: { prefix: "/baz" }
+  //     requires:
+  //       provider_name: "provider1"
+  //   - match: { prefix: "/foo" }
+  //     requires:
+  //       requires_any:
+  //         requirements:
+  //           - provider_name: "provider1"
+  //           - provider_name: "provider2"
+  //   - match: { prefix: "/bar" }
+  //     requires:
+  //       requires_all:
+  //         requirements:
+  //           - provider_name: "provider1"
+  //           - provider_name: "provider2"
+  //
+  // A special handling if rules is empty: all JWTs in a request will be verified, but the request
+  // will pass even the verifications fail. A typical usage is: this filter is used to only verify
```

Comment:
I think this is pretty surprising behavior; is there no way to design this based on principle-of-least-surprise? E.g. add an explicit "continue_request_on_verification_fail" field?

Reply:
Thanks for the input. Agreed.  I removed it and add "allow_missing_or_failed" as one of "oneof" in the JwtRequirement.